### PR TITLE
Commandline

### PR DESCRIPTION
**Pull request by tjyang**
Please review the PR again.

Two new enhancements and some minor cleanup changes
1. adding ability to get setting from brine.conf
2. adding rpm spec file for creating rpm package, work in progress.

### CODE REVIEW

File: brine.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Format see  #https://code.google.com/p/iniparse/wiki/UsageExamples
```
Comment (druonysus) [inline brine.conf]: I don't think we should have a brine.conf... I would prefer to hide the config... I think having a `.brinerc` or a maybe a `.brine` directory and then have a file called `config` in there. Personally, I like the 2nd option.

To review:

either a `.brinerc` or `.brine/config` would be the way we want to do this.

File: brine.conf [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+[brine]
+VERSION2 = 1.1
```
Comment (druonysus) [inline brine.conf]: I don't think we should version this at 1.x... the version string should be set to something like 0.0.1 to accurately represent how young of a project this is. When we do reach a point with the tool that we want to call it 1.0.0, we can tag that commit in the git repo to demarcate the official release.

File: brine.py [hunk 3]
```diff
@@ -594,26 +600,35 @@ def section_cronjobs(self):
...
     BRINEFILE  = args.brine
     if args.debug : logging.info("brine file: "+ args.brine)
-    main()
-
 
+    config = ConfigParser()     # instantiate
+    config.read('./brine.conf') # parse existing file
 
+    # Read values from a section
+    VERSION2  = config.get('brine', 'VERSION2')
```
Comment (druonysus) [inline brine.py]: I really don't like naming variables like this... maybe I need to look at the code a little closer, but likely there is a better name than VERSION2. How is it different from VERSION or VERSION1?

File: brine.py [hunk 4]
```diff
@@ -594,26 +600,35 @@ def section_cronjobs(self):
...
     brine.save_files()
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Brine '+ VERSION + 
             ' is a Salt Formula initializer. https://github.com/openx/brine')
-    parser.add_argument('-b','--brine', help='Ex: -b test01.ele ',required=True)
     parser.add_argument('-d','--debug', action='store_true',help='log debug info into brine.log', required=False)
+    parser.add_argument('-c','--conf', help='brine config file,default=./brine.conf', required=False)
+    parser.add_argument('-b','--brine', help='Ex: -b example/element/brinefile',required=True)
```
Comment (druonysus) [inline brine.py]: can we migrate `-b` and `--brine` to `-f` and `--file`? I think that feels more correct... however, I am open to other suggestions.

**Comment by tjyang** [comment]
Hi Drew

Thanks for the great comments, they have been corrected. Please review again.

```
[me@salt01 brine]$ ./brine.py -h
usage: brine.py [-h] [-d] [-c CONF] -f FILE

Brine 0.0.1 is a Salt Formula initializer. https://github.com/openx/brine

optional arguments:
  -h, --help            show this help message and exit
  -d, --debug           log debug info into brine.log
  -c CONF, --conf CONF  brine config file,default=./brine.rc
  -f FILE, --file FILE  Ex: -f example/element/brinefile
[me@salt01 brine]$ 

```

**Comment by tjyang** [comment]
pls wait for my next commit to review this PR.

**Comment by tjyang** [comment]
commits so far passed following two tests.
```
[me@salt01 brine]$ cd example/element; ../../brine.py -f brinefile -d
[me@salt01 element]$ cd -
/home/aty010/ws/brine
[me@salt01 brine]$ cd example/element; ../../brine.py -f brinefile -c ../../brine.rc -d
[me@salt01 element]$
```

File: brine.py [hunk 5]
```diff
@@ -594,26 +601,36 @@ def section_cronjobs(self):
...
     brine.save_files()
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Brine '+ VERSION + 
             ' is a Salt Formula initializer. https://github.com/openx/brine')
-    parser.add_argument('-b','--brine', help='Ex: -b test01.ele ',required=True)
     parser.add_argument('-d','--debug', action='store_true',help='log debug info into brine.log', required=False)
+    parser.add_argument('-c','--conf', help='brine config file,default=./brine.rc', required=False)
+    parser.add_argument('-f','--file', help='Ex: cd example/element; ../../brine.py -f brinefile -d',required=True)
```
Comment (druonysus) [inline brine.py]: The should not be a required option. It should look in the working directory for a file called `Brinefile` by default... if you want to point it to a different file, you can with this option.

Comment (tjyang) [inline brine.py]: pls check the latest commit to make --file as optional and reading Brinefile by default.

**Pull request merged**